src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/slurm/SlurmJobManager.groovy

```diff
@@ -1,111 +1,402 @@
+            Integer nodes = withCaughtAndLoggedException { jobResult["allocation_nodes"] as Integer }
+
+            jobInfo.usedResources = new ResourceSet(memory, cores, nodes, runTime, null, queue, null)
+            jobInfo.askedResources = new ResourceSet(null, null, null, null, null, queue, null)
+            jobInfo.runTime = runTime
+
+            /** Directories and files */
+            jobInfo.execHome = jobResult["working_directory"]
+
+            /** Timestamps */
+            jobInfo.submitTime = parseTimeOfEpochSecond(jobResult["time"]["submission"] as String)
+            jobInfo.startTime = parseTimeOfEpochSecond(jobResult["time"]["start"] as String)
+            jobInfo.endTime = parseTimeOfEpochSecond(jobResult["time"]["end"] as String)
+
+            result.put(jobID, jobInfo)
+        }
+        return result
+    }
+
+    private ZonedDateTime parseTime(String str) {
+        return withCaughtAndLoggedException {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss")
+            LocalDateTime localDateTime = LocalDateTime.parse(str, formatter)
+            return ZonedDateTime.of(localDateTime, TIME_ZONE_ID)
+        }
+    }
+
+    private ZonedDateTime parseTimeOfEpochSecond(String str) {
+        return withCaughtAndLoggedException { ZonedDateTime.ofInstant(Instant.ofEpochSecond(str as long), TIME_ZONE_ID) }
+    }
+
+    BufferValue safelyCastToBufferValue(String MAX_MEM) {
+        withCaughtAndLoggedException {
+            if (MAX_MEM) {
+                String bufferSize = MAX_MEM.find("([0-9]*[.])?[0-9]+")
+                String unit = MAX_MEM.find("[a-zA-Z]+")
+                BufferUnit bufferUnit = unit == "G" ? BufferUnit.g : BufferUnit.m
+                return new BufferValue(bufferSize, bufferUnit)
+            }
+            return null
+        }
+    }
+
+
+    @Override
+    protected ExecutionResult executeStartHeldJobs(List<BEJobID> jobIDs) {
+        String command = "scontrol release ${jobIDs*.id.join(",")}"
+        return executionService.execute(command, false)
+    }
+
+    @Override
+    ExecutionResult executeKillJobs(List<BEJobID> jobIDs) {
+        String command = "scancel ${jobIDs*.id.join(",")}"
+        return executionService.execute(command, false)
+    }
+
+    @Override
+    List<String> getEnvironmentVariableGlobs() {
+        return Collections.unmodifiableList(["SLURM_*"])
+    }
+
+    @Override
+    String getQueryJobStatesCommand() {
+        return "squeue --format='%i|%T'"
+    }
+
+    @Override
+    String getExtendedQueryJobStatesCommand() {
+        return "scontrol show job"
+    }
+
+    @Override
+    void createComputeParameter(ResourceSet resourceSet, LinkedHashMultimap<String, String> parameters) {
+        int nodes = resourceSet.isNodesSet() ? resourceSet.getNodes() : 1
+        int cores = resourceSet.isCoresSet() ? resourceSet.getCores() : 1
+        // Currently not active
+        String enforceSubmissionNodes = ''

```

Comment:
Why not delete this dead code? I doubt you have tested in anyway (how could you, it cannot be executed).